spec/requests/shared_examples/api/current/shared_spec_for_sync_requests.rb

```diff
@@ -87,4 +87,52 @@ def assert_sync_success(response, request_process_token)
 
     assert_sync_success(response, process_token)
   end
+
+  context 'resync_token in request headers is present' do
+    let(:resync_token) { "1" }
+    let(:headers_with_resync_token) { headers.merge('HTTP_X_RESYNC_TOKEN' => resync_token) }
+    let(:process_token_without_resync) { make_process_token(current_facility_processed_since: Time.now,
+                                                            other_facilities_processed_since: Time.now) }
+
+    before do
+      post sync_route, params: many_valid_records.to_json, headers: headers
+    end
+
+    it 'syncs all records from beginning if resync_token in headers is different from the one in process_token' do
+      get sync_route, params: { process_token: process_token_without_resync }, headers: headers_with_resync_token
```

Comment:
This test can be split into two; one to check when then resync_token is nil, and another when there is a mismatch

Reply:
This is already done as a separate context down below.